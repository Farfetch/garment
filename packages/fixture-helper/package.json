{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.16.0",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.16.0",
    "@garment/runner": "^0.16.0",
    "@garment/workspace": "^0.16.0",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  },
  "devDependencies": {
    "is-valid-path": "^0.1.1",
    "normalize-path": "^3.0.0"
  }
}
