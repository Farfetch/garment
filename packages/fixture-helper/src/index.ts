export * from './executeRunner';
export * from './initFixtureHelper';
export * from './resolver';
import normalizePath = require('normalize-path');
import isValidPath = require('is-valid-path');

const containsValidPath = (value: string): boolean =>
  value.split(/\s/).some((s) => isValidPath(s));

export const normalizeContainedPath = (value: string): string => {
  return value
    .split(/(\s)/)
    .map((s) => {
      if (isValidPath(s)) {
        return normalizePath(s);
      } else {
        return s;
      }
    })
    .join('');
};

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  const normalizedTestPathRegEx = new RegExp(normalizePath(testPath), 'g');
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string' && containsValidPath(value)
        ? normalizeContainedPath(value).replace(
            normalizedTestPathRegEx,
            replaceWith
          )
        : value
    )
  );
}
