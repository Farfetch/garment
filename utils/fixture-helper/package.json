{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.13.7",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.13.7",
    "@garment/runner": "^0.13.7",
    "@garment/workspace": "^0.13.7",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  }
}
