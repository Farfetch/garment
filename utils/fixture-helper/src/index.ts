export * from './executeRunner';
export * from './initFixtureHelper';

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string'
        ? value.replace(new RegExp(testPath, 'g'), replaceWith)
        : value
    )
  );
}
