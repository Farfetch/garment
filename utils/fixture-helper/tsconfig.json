{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib",
    "composite": true
  },
  "exclude": [
    "node_modules",
    "lib",
    "__tests__"
  ],
  "references": [
    {
      "path": "../../core/logger"
    },
    {
      "path": "../../core/runner"
    },
    {
      "path": "../../core/workspace"
    }
  ]
}