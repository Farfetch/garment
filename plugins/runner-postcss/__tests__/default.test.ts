import { executeRunner2, initFixtureHelper } from '@garment/fixture-helper';
import * as Path from 'path';
import handler from '../src';

const { initFixture, clean } = initFixtureHelper(module, {
  tempDir: __dirname + 'tmp__'
});

afterAll(clean);

beforeEach(async () => {
  jest.setTimeout(10000);
});

test('Uses postcss.config.js from workspace root', async () => {
  const cwd = await initFixture('basic');

  const result = await executeRunner2(handler, {} as any, {
    projectRelativePath: 'test-project',
    cwd,
    files: file => [file.text('index.css', `:fullscreen {}`)]
  });

  expect(result).toMatchSnapshot();
});

test('Uses postcss.config.js from project', async () => {
  const cwd = await initFixture('basic');

  const result = await executeRunner2(
    handler,
    { configFile: '{{projectDir}}/postcss.config.js' },
    {
      projectRelativePath: 'test-project',
      cwd,
      files: file => [
        file.text(
          'index.css',
          `
          :root {
            --color: red;
          }
          :fullscreen {color: var(--color);}
          `
        )
      ]
    }
  );

  expect(result).toMatchSnapshot();
});

test('Resolves postcss @import', async () => {
  const cwd = await initFixture('basic');

  const src = (p = '') => Path.join(cwd, 'test-project', p);

  const result = await executeRunner2(handler, {} as any, {
    projectRelativePath: 'test-project',
    cwd,
    files: () => [
      {
        path: 'test2.css',
        rootDir: src(),
        absolutePath: src('test2.css'),
        type: 'text',
        data: `
          @import './test.css'; 
          .Slideshow {
            position: relative;
            height: 50%;
        }`
      }
    ]
  });

  expect(result).toMatchSnapshot();
});
