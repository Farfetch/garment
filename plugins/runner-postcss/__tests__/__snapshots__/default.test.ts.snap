// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resolves postcss @import 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "8de170ce8b6d1c1aeb78d959f4faa0cf",
        "
          @import './test.css'; 
          .Slideshow {
            position: relative;
            height: 50%;
        }",
        "test2.css",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": ".stuff{
    background-color: aliceblue;
} 
          .Slideshow {
            position: relative;
            height: 50%;
        }",
          "path": "test2.css",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/postcss.config.js",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Transpiling files with PostCSS",
      ],
      "level": "info",
    },
    Object {
      "args": Array [
        "[PostCSS] Processing",
        "test2.css",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Uses postcss.config.js from project 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "e1c061eef28e0cce2b1ff22fb7f6e12c",
        "
          :root {
            --color: red;
          }
          :fullscreen {color: var(--color);}
          ",
        "index.css",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": "
          :fullscreen {color: red;}
          ",
          "path": "index.css",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/test-project/postcss.config.js",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Transpiling files with PostCSS",
      ],
      "level": "info",
    },
    Object {
      "args": Array [
        "[PostCSS] Processing",
        "index.css",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Uses postcss.config.js from workspace root 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "8de170ce8b6d1c1aeb78d959f4faa0cf",
        ":fullscreen {}",
        "index.css",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": ":-webkit-full-screen {}
:-ms-fullscreen {}
:fullscreen {}",
          "path": "index.css",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/postcss.config.js",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "Transpiling files with PostCSS",
      ],
      "level": "info",
    },
    Object {
      "args": Array [
        "[PostCSS] Processing",
        "index.css",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;
